galsim/integ.py

```diff
@@ -146,17 +183,27 @@ class SampleIntegrator(ImageIntegrator):
 
     @param rule         Which integration rule to apply to the wavelength and monochromatic surface
                         brightness samples.  Options include:
-                            galsim.integ.midpt  --  Use the midpoint integration rule
-                            numpy.trapz         --  Use the trapezoidal integration rule
+                            galsim.integ.midptRule  --  Use the midpoint integration rule
+                            galsim.integ.trapzRule  --  Use the trapezoidal integration rule
     """
     def __init__(self, rule):
+        if rule == midpt:
+            from galsim.deprecated import depr
+            depr('galsim.integ.midpt', 1.5, 'galsim.integ.midptRule')
+            rule = midptRule
+        elif rule == np.trapz:
+            from galsim.deprecated import depr
+            depr('galsim.integ.trapz', 1.5, 'galsim.integ.trapzRule')
```

Comment:
I think this should read `depr('np.trapz', ...`.  Also, should include a check for `if rule == galsim.inted.trapz` with this message.

Reply:
>I think this should read depr('np.trapz', .... 

Yes, good catch.  

>Also, should include a check for if rule == galsim.inted.trapz with this message.

`galsim.integ.trapz` was never an option for an `ImageIntegrator`.  It has a different syntax than `np.trapz` or `galsim.integ.midpt`.  I think Alex added it as a convenience function when he added the sampling wavelengths from `SED`s code.

Reply:
I see.  Yeah, he uses it in the tests for the SED sampling.  OK.